core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitManager.java

```diff
@@ -137,19 +157,34 @@ TwoPhaseCommitTransaction join(String txId, Isolation isolation, SerializableStr
       return resume(txId);
     }
 
-    return createNewTransaction(txId, isolation, strategy);
+    // Participant services don't use the group commit feature even if it's enabled. They simply use
```

Comment:
Sure thing!

![image](https://github.com/scalar-labs/scalardb/assets/59043/95a48a18-2d8a-4dd1-87e9-9f3a35bf2ba9)

How GroupCommitter places transactions over multiple groups depends on various timings, including some types of timeouts (e.g., moving delayed transactions to a DelayedGroup) and events (e.g., `GroupCommitter.ready(SnapShot)` is called after the `prepare` and `validate` phases are done). So, it's unlikely that the Coordinator service and all the Participant services will have the same the groups with the same transaction distributions in-memory. So, if the participant services use the group commit feature, they may insert transaction groups different from other nodes, including the Coordinator service.

See
https://github.com/scalar-labs/scalardb/blob/ebbb0274bb01fd52dc553a333aff5183434c44b6/docs/api-guide.md#commit-in-two-phase-commit-interface also.

Reply:
Understood. Thank you for your explanation!

So the way to use the two-phase interface varies depending on whether group commit is enabled or not, which might be confusing for users. One option is to always disable group commit when using the two-phase interface. What do you think?

Reply:
> So the way to use the two-phase interface varies depending on whether group commit is enabled or not, which might be confusing for users.

Yes. Users need to change the commit order in their applications as a part of the limitation, and it may be confusing.

> One option is to always disable group commit when using the two-phase interface

You mean to throw an exception or something if `TwoPhaseCommitTransactionManager` is used with the group commit feature enabled, right? Indeed, the idea sounds simpler and less confusing. On the other hand, I'm wondering what if a user wants to use the two-phase commit interface on a high-latency underlying storage where the coordinator table is placed 🤔

Reply:
> You mean to throw an exception or something if TwoPhaseCommitTransactionManager is used with the group commit feature enabled, right?

Yes, I meant that.

> On the other hand, I'm wondering what if a user wants to use the two-phase commit interface on a high-latency underlying storage where the coordinator table is placed 🤔

Correct. But, actually, there are also some other optimizations and features we have to give up for the two-phase commit interface.

Reply:
I see. That sounds a reasonable concern.

I guess it's related to the product specification. @feeblefakie Do you think we can/should prohibit the group commit feature in the two-phase commit interface?

Reply:
@brfrn169 I had a chance to talk with @feeblefakie about that and he agreed with you. I'll revise this PR.

> Users need to change the commit order in their applications as a part of the limitation

BTW, although this is a kind of memo for me about future development when we'll enable the group commit with two-phase commit interface, we should add a check whether the transaction is committed in the coordinator table,  before committing records on participant services.